tests/s3_test.go

```diff
@@ -0,0 +1,85 @@
+package tests
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/retry"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTerraformS3Bucket(t *testing.T) {
+	t.Parallel()
+
+	expectedName := fmt.Sprintf("terratest-aws-s3-example-%s", strings.ToLower(random.UniqueId()))
+
+	awsRegion := aws.GetRandomStableRegion(t, nil, nil)
+
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: "../examples/test_minimal",
+
+		Vars: map[string]interface{}{
+			"test_bucket_name": expectedName,
+			"aws_region":       awsRegion,
+		},
+
+		EnvVars: map[string]string{
+			"AWS_DEFAULT_REGION": awsRegion,
+		},
+	})
+
+	defer terraform.Destroy(t, terraformOptions)
+
+	terraform.InitAndApply(t, terraformOptions)
+
+	b, err := terraform.OutputMapOfObjectsE(t, terraformOptions, "test-bucket")
+
+	if err != nil {
+		logger.Log(t, err)
+	}
+
+	bucket_name := b["bucket_name"].(string)
+
+	assert.Equal(t, expectedName, bucket_name)
+	aws.AssertS3BucketPolicyExists(t, awsRegion, bucket_name)
+
+	// Test Unencrypted File
+	bodyString := "test-body"
+	k := fmt.Sprintf("example-file-%s", strings.ToLower(random.UniqueId()))
```

Comment:
I would expect this key to need to be in one of the read/write paths to work, based on the policy that is created:
```
  read_write_paths = ["path/to/rw-folder", "path/to/another-rw-folder"]
```

This probably works because the terratest user isn't using the policy created by the bucket to write this object.

Reply:
New commits assert policies work